sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Percentile.scala

```diff
@@ -81,7 +96,11 @@ case class Percentile(
       case arrayData: ArrayData => arrayData.toDoubleArray().toSeq
   }
 
-  override def children: Seq[Expression] = child :: percentageExpression :: Nil
+  override def children: Seq[Expression] = if (withFrqExpr) {
```

Comment:
@hvanhovell 
Can you provide your final thoughts on this and i will make the changes accordingly 
i did try to find an expression where middle argument is Optional but could not. Seems like only the last argument(s) are optional.
Which argument sequence  would you recommend 
Column [, frequency], percentages 
OR
Column, percentages [, frequency]